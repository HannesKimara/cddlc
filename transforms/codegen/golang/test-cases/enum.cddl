auth-status = &(
  authenticated: 0
  unknown-error: 1
  timeout: 2
)

close-event = {
  connection-id: uint
  reason: &(
    close-method-called: 1
    unrecoverable-error: 100
  )
}