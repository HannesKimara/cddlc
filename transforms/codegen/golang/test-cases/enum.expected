package lib

// (cddlc) Ident: AuthStatus
type AuthStatus uint

const (
	AuthStatusAuthenticated AuthStatus = 0
	AuthStatusUnknownError  AuthStatus = 1
	AuthStatusTimeout       AuthStatus = 2
)

// (cddlc) Ident: CloseEventReason
type CloseEventReason uint

const (
	CloseEventReasonCloseMethodCalled  CloseEventReason = 1
	CloseEventReasonUnrecoverableError CloseEventReason = 100
)

// (cddlc) Ident: close-event
type CloseEvent struct {
	ConnectionId uint             `cbor:"connection-id"`
	Reason       CloseEventReason `cbor:"reason"`
}
